application/models/FleetModel.php

```diff
@@ -3,69 +3,40 @@
 /**
  * Dummy model class that represent our airline's fleet
  */
-class FleetModel extends CI_Model
+class FleetModel extends CSV_Model
 {
 
-    // The mock plane data
-    var $data = array(
-        'Kkingair' => array(
-            'manufacturer' => 'Beechcraft',
-            'model' => 'King Air C90',
-            'price' => '3900',
-            'seats' => '12',
-            'reach' => '2446',
-            'cruise' => '500',
-            'takeoff' => '1402',
-            'hourly' => '990'),
-        'Kpc12ng' => array(
-            'manufacturer' => 'Pilatus',
-            'model' => 'PC-12 NG',
-            'price' => '3300',
-            'seats' => '9',
-            'reach' => '4147',
-            'cruise' => '500',
-            'takeoff' => '450',
-            'hourly' => '727'),
-        'Kcaravan' => array(
-            'manufacturer' => 'Cessna',
-            'model' => 'Grand Caravan EX',
-            'price' => '2300',
-            'seats' => '14',
-            'reach' => '1689',
-            'cruise' => '340',
-            'takeoff' => '660',
-            'hourly' => '389')
-    );
-
     // Constructor
     public function __construct()
     {
-        parent::__construct();
-
-        // inject each "record" key into the record itself, for ease of presentation
-        foreach ($this->data as $key => $record)
-        {
-            $record['key'] = $key;
-            $this->data[$key] = $record;
-        }
-    }
-
-    // Get a single plane based off its id, or null if not found
-    public function get($which)
-    {
-        return !isset($this->data[$which]) ? null : $this->data[$which];
+        parent::__construct(APPPATH . DATA_FLEET, 'key');
     }
 
     // Get all the planes
-    public function all()
+    public function allAsArray()
     {
-        return $this->data;
+        $fleet = array();
+
+        foreach ($this->all() as $plane)
+        {
+            $fleet[$plane->key] = (array)$plane;
+        }
+
+        return $fleet;
     }
 
-    // Get number of planes in fleet
-    public function size()
+    // Get a single plane based off its id, or null if not found
+    public function getPlane($id)
```

Comment:
Is this needed? I don't think this function is used anywhere. 
Also isn't this functionality done by the `get` function in `Memory_Model` class?

Reply:
ya you're right it should. removed